apps/jubmoji-quest/src/pages/powers/[id].tsx

```diff
@@ -0,0 +1,218 @@
+import AppHeader from "@/components/AppHeader";
+import { Icons } from "@/components/Icons";
+import { PowerCard } from "@/components/cards/PowerCard";
+import { Button } from "@/components/ui/Button";
+import { JubmojiPower } from "@/types";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+import { $Enums } from "@prisma/client";
+import { useJubmojis } from "@/hooks/useJubmojis";
+import {
+  Jubmoji,
+  JubmojiInCollection,
+  JubmojiInCollectionWithNonce,
+  NUniqueJubmojisInCollection,
+  createProofInstance,
+  getCardPubKeyFromIndex,
+} from "jubmoji-api";
+import { Prisma } from "@prisma/client";
+
+export type QRCodePowerProps = {
+  power: JubmojiPower;
+  jubmojis: Jubmoji[];
+};
+
+const QRCodePower = ({ power, jubmojis }: QRCodePowerProps) => {
+  const onUsePower = async () => {
+    alert("Using power...");
+
+    switch (power.quest.proofType) {
+      case $Enums.ProofType.IN_COLLECTION:
```

Comment:
This looks like a ton of repeated code per case; can we factor some stuff out?

Reply:
factored